Fix multiple "description must start with capital letter" audit errors

issyl0 opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

- This is **WIP** while I go through the `brew audit --strict` list in its entirety for this "description must start with a capital letter" issue, build all these changes locally and audit them again, but I wanted to open it to get my approach validated and to make others aware I'm working on fixing this particular linting error.
- Are WIP PRs allowed? Is one commit per changed formula the proper way to do it (I've seen big PRs like this elsewhere in `homebrew-core`). Please tell me if it's not the right way to do things!

MikeMcQuaid commented:
> Are WIP PRs allowed? Is one commit per changed formula the proper way to do it (I've seen big PRs like this elsewhere in homebrew-core). Please tell me if it's not the right way to do things!

Yup! This is great, thanks.

issyl0 commented:
I'm done with fixing them now (the ones that make sense to fix - add "a" or "an" in front, capitalise the first word, or reword the description), just need to test everything works still.

ilovezfs commented:
There's another audit rule that the description must not start with A or An.

issyl0 commented:
@ilovezfs Yep - I just found that out the hard way. I'll think of other ways to fix those.

ilovezfs commented:
Probably just for the sake of consistency. Also, Zfs should be spelled ZFS.

ilovezfs commented on Formula/znapzend.rb:
```diff
@@ -1,5 +1,5 @@
 class Znapzend < Formula
-  desc "zfs backup with remote capabilities and mbuffer integration"
+  desc "Zfs backup with remote capabilities and mbuffer integration"
```
here

issyl0 commented:
Edited to not start with "A" or "An", and `brew audit --strict` seems to be happy locally, so I'm removing the WIP.

issyl0 commented:
I'll wait until https://github.com/Homebrew/brew/pull/2748 is merged and then go through this again to revert my "MacOS" and "based on xUnit" changes which were very forced. I'm also not sure why CI aborted, and I was hesitant to run it again in case it was deliberate.

MikeMcQuaid commented:
@issyl0 Any news on this? No worries if not yet!

issyl0 commented:
@MikeMcQuaid Sorry, life got in the way. I'm working on this now. :-)

issyl0 commented:
I've rebased on master and deleted some commits (four in total) that "fixed" things that as of https://github.com/Homebrew/brew/pull/2748/commits/c572081f8b56f3be24f24e85497ae780a25cbcba shouldn't trigger the linter.

MikeMcQuaid commented on Formula/aap.rb:
```diff
@@ -1,5 +1,5 @@
 class Aap < Formula
-  desc "make-like tool to download, build, and install software"
+  desc "Tool similar to make to download, build, and install software"
```
I think this could be `Make-like tool`. Generally any case where it gets reordered to avoid a name being uppercased: let me know as it should probably be added to the exception list.

MikeMcQuaid commented on Formula/bash-preexec.rb:
```diff
@@ -1,5 +1,5 @@
 class BashPreexec < Formula
-  desc "preexec and precmd functions for Bash just like Zsh."
```
Again, I reckon a whitelist of this name may make sense. I'd put the `(just like Zsh)` because the reordering changes the meaning a bit.

MikeMcQuaid replied:
It's also just occurred to me that we should probably enforce the lack of full-stops at the end of descriptions. What do you think?

issyl0 replied:
Enforcing the lack of full stops sounds good, but this one does seem to be an anomaly - running `ack "desc\b" *.rb | ack "\.$" *.rb` returned nothing else (though this branch is several days old by now).

issyl0 replied:
I just rebased on top of the latest master, and the result of those searches is still the same.

issyl0 replied:
I know there are none now, but in the interests of consistency (so that none pop up again), I suppose we probably should. https://github.com/Homebrew/brew/pull/2795

ilovezfs replied:
Not sure it makes sense to add audits for things that don't actually exist in the wild out of speculation.

ilovezfs replied:
However, I think it does exist in the wild ...

```
Josephs-MacBook-Pro:Formula joe$ grep -r -E 'desc ".*\."' .|wc -l
     133
```

issyl0 replied:
True. I guess I'm just enjoying contributing too much. 🤗

ilovezfs replied:
```
ack "\.$" *.rb 
```
They would end with ." not just .

issyl0 replied:
Ah yes, the `"`. Maybe half past seven in the morning isn't the best time to be thinking about regexes. 😴

MikeMcQuaid commented on Formula/docker-compose-completion.rb:
```diff
@@ -1,5 +1,5 @@
 class DockerComposeCompletion < Formula
-  desc "docker-compose completion script"
+  desc "Completion script for docker-compose"
```
Similarly, let's avoid reordering here.

MikeMcQuaid commented on Formula/dwarfutils.rb:
```diff
@@ -1,5 +1,5 @@
 class Dwarfutils < Formula
-  desc "lib and utility to dump and produce DWARF debug info in ELF objects"
+  desc "Lib and utility to dump and produce DWARF debug info in ELF objects"
```
`Library` instead of `Lib`

issyl0 replied:
I seem to remember trying that in the first place, but it hit the "80 characters" rule, and I didn't know how to shorten it coherently.

MikeMcQuaid replied:
Could just go for `Dump and produce DWARF debug information in ELF objects`

MikeMcQuaid commented on Formula/gotags.rb:
```diff
@@ -1,5 +1,5 @@
 class Gotags < Formula
-  desc "ctags-compatible tag generator for Go"
+  desc "Tag generator for Go, compatible with ctags"
```
This reordering seems better actually

MikeMcQuaid commented on Formula/jemalloc.rb:
```diff
@@ -1,5 +1,5 @@
 class Jemalloc < Formula
-  desc "malloc implementation emphasizing fragmentation avoidance"
+  desc "Implementation of malloc emphasizing fragmentation avoidance"
```
Again, what about whitelisting `malloc`

MikeMcQuaid commented on Formula/ksh.rb:
```diff
@@ -1,6 +1,6 @@
 
 class Ksh < Formula
-  desc "ksh93, the KornShell"
+  desc "The KornShell, ksh93"
```
`KornShell, ksh93` seems fine

MikeMcQuaid commented on Formula/neatvi.rb:
```diff
@@ -1,5 +1,5 @@
 class Neatvi < Formula
-  desc "ex/vi clone for editing bidirectional uft-8 text"
+  desc "Clone of ex/vi for editing bidirectional uft-8 text"
```
How about we whitelist this?

MikeMcQuaid commented on Formula/rock.rb:
```diff
@@ -1,5 +1,5 @@
 class Rock < Formula
-  desc "ooc compiler written in ooc"
+  desc "Compiler for occ, written in ooc"
```
How about we whitelist `ooc`?

MikeMcQuaid commented on Formula/ucg.rb:
```diff
@@ -1,5 +1,5 @@
 class Ucg < Formula
-  desc "grep-like tool for searching large bodies of source code"
+  desc "Tool for searching large bodies of source code, like grep"
```
I wonder if the `, like $FOO` are better as `(like $FOO)`. What do you think?

MikeMcQuaid commented on Formula/xar-mackyle.rb:
```diff
@@ -1,5 +1,5 @@
 class XarMackyle < Formula
-  desc "eXtensible ARchiver"
+  desc "The eXtensible ARchiver"
```
I think we can whitelist this one.

issyl0 commented:
Thanks, Mike. The whitelisting sounds good, if you're happy with whitelisting all those - there seems to be a lot! Shall I raise a PR for the whitelist?

MikeMcQuaid commented:
> Shall I raise a PR for the whitelist?

Yes, that would be fantastic, thanks!

issyl0 commented:
There's a PR at https://github.com/Homebrew/brew/pull/2780 that adds "ex, eXtensible, malloc, ooc and preexec" to the whitelist.

MikeMcQuaid commented:
> There's a PR at Homebrew/brew#2780 that adds "ex, eXtensible, malloc, ooc and preexec" to the whitelist.

@issyl0 Merged! @mention me when you've rebased this and I'll try to merge this ASAP.

issyl0 commented:
@MikeMcQuaid This is ready now. Thanks for your patience!

ilovezfs commented on Formula/bash-preexec.rb:
```diff
@@ -1,5 +1,5 @@
 class BashPreexec < Formula
-  desc "preexec and precmd functions for Bash just like Zsh."
+  desc "preexec and precmd functions for Bash (just like ZSH)"
```
I don't think ZSH is usually all caps.

issyl0 replied:
Oops, changed.

ilovezfs commented on Formula/path-extractor.rb:
```diff
@@ -1,5 +1,5 @@
 class PathExtractor < Formula
-  desc "unix filter which outputs the filepaths found in stdin"
+  desc "Unix filter which outputs the filepaths found in stdin"
```
I believe it's usually UNIX in other descriptions

issyl0 replied:
So it is. Changed.

Thanks for catching these.

ilovezfs commented:
Thanks for the careful work here @issyl0!

MikeMcQuaid commented:
Thanks again @issyl0!
